src/modules/SearchPage/index.tsx

```diff
@@ -1,67 +1,151 @@
-import { fetchDetailedPokemons, fetchPokemons } from 'utils';
-import { API_URL, INPUT_VALUE_KEY } from 'appConstants';
-import { SearchBar, Card } from './components';
-import React, { ChangeEvent } from 'react';
-import { PokemonDetails } from 'types';
-import { Loader } from 'components';
+import { AvailableTabs, API, INPUT_VALUE_KEY, tabs } from 'appConstants';
+import { fetchPokemonByParameter, fetchDetails, fetchBase } from 'utils';
+import { SearchBar, Tabs, Cards } from 'modules/SearchPage/components';
+import React, { ChangeEvent, KeyboardEvent } from 'react';
 import { Layout } from 'modules';
 import * as S from './styled';
+import {
+  PokemonTypeDetails,
+  PokemonMoveDetails,
+  PokemonDetails,
+  PokemonData,
+  PokemonMove,
+  PokemonType,
+  MovesData,
+  TypesData,
+  Pokemon,
+  Lengths,
+} from 'types';
 
 type Props = {
   componentName: string;
   location: string;
 };
 
 type State = {
+  moves: Array<PokemonMoveDetails>;
+  types: Array<PokemonTypeDetails>;
   pokemons: Array<PokemonDetails>;
+  activeTab: AvailableTabs;
   inputValue: string;
   isLoading: boolean;
+  lengths: Lengths;
 };
 
 export class SearchPage extends React.Component<Props, State> {
   constructor(props: Props) {
     super(props);
     this.state = {
+      activeTab: AvailableTabs.pokemons,
       isLoading: true,
       inputValue: '',
       pokemons: [],
+      moves: [],
+      types: [],
+      lengths: {
+        moves: 0,
+        pokemons: 0,
+        types: 0,
+      },
     };
   }
+
   componentDidMount() {
-    const value = window.localStorage.getItem(INPUT_VALUE_KEY);
-    if (value != null) {
-      this.setState({ inputValue: value });
+    const inputValue = window.localStorage.getItem(INPUT_VALUE_KEY);
+    if (inputValue && inputValue.trim().length > 0) {
+      this.setState({ inputValue });
+      this.fetchAndSetSpecificData(inputValue).then(() =>
+        setTimeout(() => this.setState({ isLoading: false }), 500)
+      );
+    } else {
+      this.fetchAndSetAllData().then(() =>
+        setTimeout(() => this.setState({ isLoading: false }), 500)
+      );
     }
-    fetchPokemons(API_URL)
-      .then((pokemons) => fetchDetailedPokemons(pokemons))
-      .then((pokemons) => this.setState({ pokemons }))
-      .finally(() => setTimeout(() => this.setState({ isLoading: false }), 1000));
   }
-  onChange = (event: ChangeEvent<HTMLInputElement>) => {
+
+  onClick = (tab: AvailableTabs) => this.setState({ activeTab: tab });
+
+  onChange = async (event: ChangeEvent<HTMLInputElement>) => {
     const { value } = event.target;
     this.setState({ inputValue: value });
     window.localStorage.setItem(INPUT_VALUE_KEY, value);
   };
+
+  onKeyDown = async ({ key }: KeyboardEvent<HTMLInputElement>) => {
+    if (key === 'Enter') {
+      this.setState({ isLoading: true });
+      const value = this.state.inputValue;
+      if (value.trim().length > 0) await this.fetchAndSetSpecificData(value);
```

Comment:
!!value.trim().length 
and also no more magic numbers  ;-)

Reply:
done